localstack/aws/handlers/partition_rewriter.py

```diff
@@ -0,0 +1,219 @@
+
+    arn_regex = re.compile(
+        r"arn:"  # Prefix
+        r"(?P<Partition>(aws|aws-cn|aws-iso|aws-iso-b|aws-us-gov)*):"  # Partition
+        r"(?P<Service>[\w-]*):"  # Service (lambda, s3, ecs,...)
+        r"(?P<Region>[\w-]*):"  # Region (us-east-1, us-gov-west-1,...)
+        r"(?P<AccountID>[\w-]*):"  # AccountID
+        r"(?P<ResourcePath>"  # Combine the resource type and id to the ResourcePath
+        r"((?P<ResourceType>[\w-]*)[:/])?"  # ResourceType (optional, f.e. S3 bucket name)
+        r"(?P<ResourceID>[\w\-/*]*)"  # Resource ID (f.e. file name in S3)
+        r")"
+    )
+
+    def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
+        request = context.request
+        # If this header is present, or the request is internal, remove it and continue the handler chain
+        if request.headers.pop("LS-INTERNAL-REWRITE-HANDLER", None) or is_internal_call_context(
+            request.headers
+        ):
+            return
+        # since we are very early in the handler chain, we cannot use the request context here
+        request_region = extract_region_from_headers(request.headers)
+        forward_request = self.modify_request(request)
+        # forward to the handler chain again
+        result_response = forward(
+            request=forward_request,
+            forward_base_url=config.get_edge_url(),
+            forward_path=forward_request.path,
+            headers=forward_request.headers,
+        )
+        self.modify_response(result_response, request_region=request_region)
+        response.update_from(result_response)
+
+        # terminate this chain, as the request was proxied
+        chain.terminate()
+
+    def modify_request(self, request: Request) -> Request:
+        """
+        Modifies the request by rewriting ARNs
+
+        :param request: Request
+        :return: New request with rewritten data
+        """
+        # rewrite inbound request
+        forward_rewritten_path, forward_rewritten_query_string = self._adjust_partition_in_path(
+            request.full_path, self.DEFAULT_INBOUND_PARTITION
+        )
+        forward_rewritten_body = self._adjust_partition(
+            restore_payload(request), self.DEFAULT_INBOUND_PARTITION
+        )
+        forward_rewritten_headers = self._adjust_partition(
+            dict(request.headers), self.DEFAULT_INBOUND_PARTITION
+        )
+
+        # add header to signal request has already been rewritten
+        forward_rewritten_headers["LS-INTERNAL-REWRITE-HANDLER"] = "1"
+        # Create a new request with the updated data
+        return Request(
+            method=request.method,
+            headers=forward_rewritten_headers,
+            path=forward_rewritten_path,
+            body=forward_rewritten_body,
+            # we have to set query string to None to avoid it being counted as defined in werkzeug
+            query_string=forward_rewritten_query_string,
+        )
```

Comment:
as you pointed out, doing the re-writing in this way is problematic because it blocks any streamed bodies. maybe it would be good to skip the arn rewriting for bodies that are streamed?

what you can do is to check 
```python
if context.operation.has_streaming_input:
    ...
```
and in that case, skip the re-writing, and to the new Request's body, pass `request._get_stream_for_parsing()` (i know that's not great, it's what we do in the parser:
https://github.com/localstack/localstack/blob/0aa9ffb17d6e0e76b0cab5f7f9ddca45bc46c685/localstack/aws/protocol/parser.py#L592-L597
https://github.com/localstack/localstack/blob/0aa9ffb17d6e0e76b0cab5f7f9ddca45bc46c685/localstack/aws/protocol/parser.py#L634-L644

basically this should solve the proxying issue, assuming that no other handler before that has consumed the stream. i'm not 100% sure this will work that way though tbh.